pkg/fanal/types/package.go

```diff
@@ -176,6 +176,22 @@ type BuildInfo struct {
 	Arch        string   `json:",omitempty"`
 }
 
+// RepositoryClass indicates the class of repository from which the package was distributed.
+type RepositoryClass string
+
+const (
+	RepositoryClassUnknown    RepositoryClass = ""            // Unknown or not set
+	RepositoryClassOfficial   RepositoryClass = "official"    // Official OS repository (Debian, Ubuntu, Red Hat, etc.)
+	RepositoryClassThirdParty RepositoryClass = "third-party" // Third-party repository (Docker, NVIDIA, etc.)
+)
+
+// PackageRepository indicates where the package was distributed from.
+type PackageRepository struct {
```

Comment:
idea for another PR:
We can consider adding a new field for APK.

Currently, we use the `Repository` field in `AnalysisResult`:
https://github.com/aquasecurity/trivy/blob/8bf6caf98e2b1eff7bd16987f6791122d827747c/pkg/fanal/analyzer/repo/apk/apk.go#L80-L86

However, I think we can deprecate and eventually remove `AnalysisResult.Repository`,
and instead store repository information at the package level.